src/ape_run/_cli.py

```diff
@@ -115,14 +115,16 @@ def commands(self) -> Dict[str, Union[click.Command, click.Group]]:
             return {}
 
         commands = {}
-        for filepath in self._project.scripts_folder.glob("*.py"):
-            if filepath.stem.startswith("_"):
-                continue  # Ignore any "private" files
-
-            cmd = self._get_command(filepath)
-            if cmd:  # NOTE: Don't allow calling commands that failed to load
-                commands[filepath.stem] = cmd
-
+        scripts_folders = [x for x in self._project.scripts_folder.iterdir() if x.is_dir()]
```

Comment:
So we don't need to gather the dirs here. We need to parse the user input and find the script. For example, if the users does:

```
ape run sequencer.middleware.build
```

, we would parse that out and figure out that they are trying to run the script `sequencer/middleware/build.py`
So we don't need to gather all the directories, we need to figure out the to the script they want.

**Also, this allows us to have scripts with the same name across different sub-directories as well as n-level deep directories.**

Reply:
Correction:

```
ape run sequencer middleware build
```

as click helps us a lot; the subdirs can be Groups.

Reply:
> Correction:
> 
> ```
> ape run sequencer middleware build
> ```
> 
> as click helps us a lot; the subdirs can be Groups.

Yoooo, this is sick

I love this optimization of turning them into groups